aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -948,9 +948,20 @@ static Tensor make_offset2bag(
     const Tensor& offsets,
     const int64_t mode,
     const c10::optional<Tensor>& per_sample_weights,
-    const int64_t padding_idx) {
+    const int64_t padding_idx,
+    bool include_last_offset = false) {
   Tensor offset2bag = at::empty({0}, offsets.options());
-  make_offset2bag_out(offset2bag, output, weight, indices, offsets, mode, per_sample_weights, padding_idx);
+  // when include_last_offset is true, ignore the last index in offset.
+  //
+  // fix runtime erro when include_last_offset is true and offsets[-1] != indices.size(0)
+  // see https://github.com/pytorch/pytorch/issues/89677 for more details.
+  //
+  Tensor _offsets = offsets;
+  if (include_last_offset) {
+    _offsets = offsets.clone();
+    _offsets[-1] = indices.size(0);
```

Comment:
I think I'd very much like to avoid having to clone offsets here. Is there a way to adjust make_offset2bag_out to handle this case directly?

Reply:
Good idea, i have made the change in `make_offset2bag`: when `include_last_offset` is true, the last index from `offsets` is treated as `indices.size(0)` (no matter what the actual value is).